subprojects/groovy-jsr223/src/main/java/org/codehaus/groovy/jsr223/GroovyScriptEngineImpl.java

```diff
@@ -275,6 +238,19 @@ public Object getVariable(String name) {
                     if (scope != -1) {
                         return ctx.getAttribute(name, scope);
                     }
+                    // Redirect script output to context writer, if out var is not already provided
+                    if ("out".equals(name)) {
+                        Writer writer = ctx.getWriter();
+                        if (writer != null) {
+                            return (writer instanceof PrintWriter) ?
+                                    (PrintWriter) writer :
+                                    new PrintWriter(writer, true);
```

Comment:
This could potentially create lots of temporary `PrintWriter` objects if `ctx.getWriter()` returns a different type.  But the alternative of caching the writer may mean keeping a reference to it longer than necessary.